net/tsaddr/tsaddr.go

```diff
@@ -164,3 +164,24 @@ func NewContainsIPFunc(addrs []netaddr.IPPrefix) func(ip netaddr.IP) bool {
 	}
 	return func(ip netaddr.IP) bool { return m[ip] }
 }
+
+// PrefixesContainsFunc reports whether f is true for any IPPrefix in
+// ipp.
+func PrefixesContainsFunc(ipp []netaddr.IPPrefix, f func(netaddr.IPPrefix) bool) bool {
```

Comment:
aren't we getting generics for this? :-P

Reply:
SOON